examples/text-generation/run_generation.py

```diff
@@ -52,9 +52,15 @@ def generate(model, tokenizer, prompts, length, temperature):
         default="One of my fondest memory is",
         help="The prompts to use for generation, using | as separator.",
     )
+    parser.add_argument(
+        "--batch_size",
+        type=int,
+        default=None,
+        help="The batch size (optional). If not specified it wil be based on the number of prompts.",
+    )
     parser.add_argument("--length", type=int, default=128, help="The number of tokens in the generated sequences.")
     parser.add_argument(
-        "--num_cores", type=int, default=2, help="The number of cores on which the model should be split."
+        "--num_cores", type=int, default=1, help="The number of cores on which the model should be split."
     )
     parser.add_argument("--auto_cast_type", type=str, default="f32", help="One of f32, f16, bf16.")
```

Comment:
```suggestion
    parser.add_argument("--auto_cast_type", type=str, default="f32", choices=["f32", "f16", "bf16"], help="One of f32, f16, bf16.")
```
Also I would argue that it would be better if it was `fp32`, `fp16` and `bf16` as it is more aligned with PyTorch.

Reply:
These are the values used by the neuron compiler. I added the choices though.